tests/nn/data_parallel/test_fsdp_freezing_weights.py

```diff
@@ -99,10 +103,14 @@ def _distributed_worker(
     model = model.cuda()
 
     # freezing the trunk using requires_grad.
-    if freezing_method == FreezingMethod.RequiresGrad:
+    if False and freezing_method == FreezingMethod.RequiresGrad:
         for param in model.trunk.parameters():
             param.requires_grad = False
 
+    # Freeze trunk2.
+    for param in model.trunk2.parameters():
```

Comment:
freeze after wrapped.

Reply:
For education purpose, what does freeze mean? What is the purpose of freeze?

Reply:
Very good question. Thanks for asking!

In neural network training, the parameters are updated (optimized) by learning from the data and the losses. Freezing means we want to keep certain parameters fixed without them being updated or optimized by this process. This allows the rest of the parameters to learn and keep some part of the model fixed. This is a common thing to do in training.